apps/web/src/store/collateral/hooks.ts

```diff
@@ -2,42 +2,55 @@ import React, { useMemo } from 'react';
 
 import { addresses, CallData } from '@balancednetwork/balanced-js';
 import BigNumber from 'bignumber.js';
-import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { keepPreviousData, useQuery, UseQueryResult } from '@tanstack/react-query';
 import { useDispatch, useSelector } from 'react-redux';
 
 import bnJs from 'bnJs';
-import { NETWORK_ID } from 'constants/config';
+import { ICON_XCALL_NETWORK_ID, NETWORK_ID } from 'constants/config';
 import { MINIMUM_ICX_FOR_ACTION } from 'constants/index';
-import { NULL_CONTRACT_ADDRESS, SUPPORTED_TOKENS_MAP_BY_ADDRESS } from 'constants/tokens';
-import { useBorrowedAmounts, useLoanParameters, useLockingRatios } from 'store/loan/hooks';
-import { useOraclePrice, useOraclePrices } from 'store/oracle/hooks';
+import { SUPPORTED_TOKENS_LIST } from 'constants/tokens';
+import { useBorrowedAmounts } from 'store/loan/hooks';
+import { useOraclePrice } from 'store/oracle/hooks';
 import { useRatio } from 'store/ratio/hooks';
 import { useAllTransactions } from 'store/transactions/hooks';
-import { useICONWalletBalances } from 'store/wallet/hooks';
+import { useCrossChainWalletBalances, useICONWalletBalances } from 'store/wallet/hooks';
 import { CurrencyKey, IcxDisplayType } from 'types';
-import { formatUnits, toBigNumber } from 'utils';
+import { formatUnits, maxAmountSpend, toBigNumber } from 'utils';
 
 import { AppState } from '../index';
 import {
   adjust,
   cancel,
   changeDepositedAmount,
   changeCollateralType,
+  changeCollateralXChain,
   changeIcxDisplayType,
   type,
   Field,
 } from './reducer';
+import { Position, XChainId, XCollaterals, XPositionsRecord, XToken } from 'app/pages/trade/bridge/types';
+import { DEFAULT_TOKEN_CHAIN, xTokenMap } from 'app/pages/trade/bridge/_config/xTokens';
+import { useSignedInWallets, useAvailableWallets } from 'app/pages/trade/bridge/_hooks/useWallets';
+import { Currency, CurrencyAmount } from '@balancednetwork/sdk-core';
+import { SUPPORTED_XCALL_CHAINS, xChainMap } from 'app/pages/trade/bridge/_config/xChains';
+import { setRecipientNetwork } from 'store/loan/reducer';
+import { useDestinationEvents } from 'app/pages/trade/bridge/_zustand/useXCallEventStore';
+import { forEach } from 'lodash-es';
 import { useRatesWithOracle } from 'queries/reward';
 import { getBalanceDecimals } from 'utils/formatter';
 
 export const DEFAULT_COLLATERAL_TOKEN = 'sICX';
 
-export function useCollateralChangeDepositedAmount(): (depositedAmount: BigNumber, token?: string) => void {
+export function useCollateralChangeDepositedAmount(): (
```

Comment:
add the function in `useCollateralActionHandlers` hook and change the function name to `changeDepositedAmount`

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388